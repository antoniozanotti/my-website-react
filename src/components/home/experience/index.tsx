import Section from "@/components/ui/section";
import WorksList from "@/components/common/work-list";
import { useTranslations } from "next-intl";

export default function Index({ positions }: { positions: any }) {
  const t = useTranslations('Experience');
  return (
    <Section title={t('title')}>
      <WorksList
        works={positions}
        labelForMore={t("labelForMore")}
      />
    </Section>
  );
}
