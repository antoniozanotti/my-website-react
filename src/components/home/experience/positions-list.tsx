"use client";
import { useState } from "react";
import Position from "./position";
import { TzButton } from "topaz-react";
import type { VersionType } from "@/const/version";
import versionType from "@/const/version";

export default function PositionsList({ positions }: { positions: any }) {
  const [version, setVersion] = useState<VersionType>(versionType.SHORT);
  positions.map((position: any, index: number) => {
    if (index == 7) {
      console.log(position.achievements.json.content);
    }
  });
  return (
    <article>
      {positions.map(
        (position: any, index: number) =>
          ((version == versionType.SHORT && position.visibleInShortVersion) ||
            version == versionType.FULL) && (
            <Position
              key={index}
              title={position.title}
              companyName={position.company}
              location={position.location}
              period={position.period}
              achievements={position.achievements.json.content
                .filter((a: any) => a.content[0].value != "")
                .map((achievement: any) => {
                  return achievement.content[0].value;
                })}
              links={position.links?.json.content
                .filter((link: any) => link.nodeType == "unordered-list")[0]
                .content.map((link: any) => {
                  return {
                    link: link.content[0].content.filter(
                      (l: any) => l.nodeType == "hyperlink"
                    )[0].data.uri,
                    label: link.content[0].content.filter(
                      (l: any) => l.nodeType == "hyperlink"
                    )[0].content[0].value,
                  };
                })}
              skills={position.skillsCollection.items.map(
                (item: any) => {
                  return item.title;
                }
              )}
            />
          )
      )}
      {version == versionType.SHORT && (
        <aside className="flex gap-2 mt-[48px]">
          <TzButton
            iconName="ChevronDoubleDownIcon"
            isIconAfterLabel
            variant="accent"
            label="See full employment history"
            onClick={() => setVersion(versionType.FULL)}
          />
        </aside>
      )}
    </article>
  );
}
