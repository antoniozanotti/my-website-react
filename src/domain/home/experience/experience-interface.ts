import { ArticleType } from "@/types/article-type";

export interface ExperienceInterface {
  positions: ArticleType[];
}
