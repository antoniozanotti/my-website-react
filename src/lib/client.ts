import { ApolloClient, HttpLink, InMemoryCache } from "@apollo/client";
import { registerApolloClient } from "@apollo/experimental-nextjs-app-support/rsc";

const TOKEN = process.env.CONTENTFUL_ACCESS_TOKEN;
const SPACE = process.env.CONTENTFUL_SPACE_ID;
const ENVIRONMENT = process.env.CONTENTFUL_ENVIRONMENT_ID;
const URL = `https://graphql.contentful.com/content/v1/spaces/${SPACE}/environments/${ENVIRONMENT}`;

export const { getClient } = registerApolloClient(() => {
  return new ApolloClient({
    cache: new InMemoryCache(),
    link: new HttpLink({
      uri: URL,
      headers: {
        Authorization: `Bearer ${TOKEN}`,
      },
    }),
  });
});
