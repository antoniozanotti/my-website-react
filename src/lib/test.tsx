import React, { ReactElement } from "react";
import {
  render,
  RenderOptions,
  RenderResult,
  Screen,
} from "@testing-library/react";
import { Theme } from "@/domain/theme/theme/theme";
import { Providers } from "../app/providers";
import messages from "@/messages/en.json";

function AllTheProviders({ children }: { children: React.ReactNode }) {
  return (
    <Providers locale="en" messages={messages}>
      <Theme>
        <div data-testid="container">{children}</div>
      </Theme>
    </Providers>
  );
}

function customRender(
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
): RenderResult {
  return render(ui, { wrapper: AllTheProviders, ...options });
}

function firstComponent(screen: Screen) {
  return screen.getByTestId("container").firstChild;
}

export * from "@testing-library/react";
export * from "@testing-library/user-event";
export { customRender as render, firstComponent };
import "@testing-library/jest-dom";

jest.mock("next/navigation", () => {
  return {
    useRouter() {
      return { push: () => {} };
    },
    useParams: () => ({}),
  };
});
