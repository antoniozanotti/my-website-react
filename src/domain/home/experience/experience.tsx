import { ExperienceInterface } from "./experience-interface";
import { useTranslations } from "next-intl";
import { Section } from "@/domain/ui/section/section";
import { ArticleListContainer } from "../article-list/article-list-container";

export function Experience({ positions }: ExperienceInterface) {
  const t = useTranslations('Experience');
  
  return (
    <Section title={t('title')}>
      <ArticleListContainer
        articles={positions}
        labelForMore={t("labelForMore")}
      />
    </Section>
  );
}
