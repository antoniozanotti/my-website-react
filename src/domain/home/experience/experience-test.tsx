import { screen, render } from "@/lib/test";
import { Experience } from "./experience";
import { ExperienceMocks } from './experience-mocks';

describe("Experience", () => {
  test("should render section", async () => {
    render(<Experience positions={ExperienceMocks.positions} />);
    const title = screen.getByRole("heading", {name: "Experience"});
    
    expect(title).toBeVisible();
  });
});
