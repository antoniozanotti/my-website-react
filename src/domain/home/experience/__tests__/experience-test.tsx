import { screen, render } from "@/lib/test";
import { Experience } from "../experience";
import { ExperienceMocks } from './experience-mocks';

describe("Experience", () => {
  it("should render section", async () => {
    const { container } = render(<Experience positions={ExperienceMocks.positions} />);
    const title = screen.getByRole("heading", {name: "Experience"});
    
    expect(title).toBeVisible();
    expect(container).toMatchSnapshot();
  });
});
