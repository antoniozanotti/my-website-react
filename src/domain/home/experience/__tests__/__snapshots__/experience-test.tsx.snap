// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Experience should render section 1`] = `
<div>
  <div
    class="dark"
  >
    <div
      class="className bg-dark-2 dark:bg-dark-dark-2 text-primary dark:text-dark-primary text-c3 lg:text-c4"
    >
      <div
        data-testid="container"
      >
        <section>
          <h2
            class="leading-none text-c9 sm:text-c10 lg:text-c11 text-accent-1 dark:text-dark-accent-1 drop-shadow mt-[164px] "
            id=""
          >
            Experience
          </h2>
          <div>
            <div>
              <div
                style="opacity: 0;"
              >
                <h3
                  class="leading-none text-c7 sm:text-c8 lg:text-c10 text-accent-2 dark:text-dark-accent-2 drop-shadow mt-[68px] "
                  id=""
                >
                  Full Stack Developer
                </h3>
                <div
                  class="sm:grid sm:gap-x-8 lg:gap-x-16 sm:grid-cols-3 mt-4 space-y-4"
                >
                  <div
                    class="grid grid-flow-row auto-rows-max"
                  >
                    <span>
                      MAD Design Group
                    </span>
                    <span
                      class="italic text-c2 lg:text-c3"
                    >
                      Sydney, New South Wales, Australia
                    </span>
                    <span
                      class="font-bold py-4"
                    >
                      Jul 2015 - Dec 2023
                    </span>
                    <div
                      class="flex flex-wrap gap-2"
                    >
                      <span
                        class="rounded font-medium px-2 py-1 text-c2  bg-secondary dark:bg-dark-secondary text-inverse-secondary dark:text-inverse-dark-secondary "
                      >
                        Front-End Development
                      </span>
                      <span
                        class="rounded font-medium px-2 py-1 text-c2  bg-secondary dark:bg-dark-secondary text-inverse-secondary dark:text-inverse-dark-secondary "
                      >
                        Software Architecture
                      </span>
                      <span
                        class="rounded font-medium px-2 py-1 text-c2  bg-secondary dark:bg-dark-secondary text-inverse-secondary dark:text-inverse-dark-secondary "
                      >
                        React
                      </span>
                      <span
                        class="rounded font-medium px-2 py-1 text-c2  bg-secondary dark:bg-dark-secondary text-inverse-secondary dark:text-inverse-dark-secondary "
                      >
                        Next.js
                      </span>
                      <span
                        class="rounded font-medium px-2 py-1 text-c2  bg-secondary dark:bg-dark-secondary text-inverse-secondary dark:text-inverse-dark-secondary "
                      >
                        Node.js
                      </span>
                      <span
                        class="rounded font-medium px-2 py-1 text-c2  bg-secondary dark:bg-dark-secondary text-inverse-secondary dark:text-inverse-dark-secondary "
                      >
                        TypeScript
                      </span>
                      <span
                        class="rounded font-medium px-2 py-1 text-c2  bg-secondary dark:bg-dark-secondary text-inverse-secondary dark:text-inverse-dark-secondary "
                      >
                        JavaScript
                      </span>
                      <span
                        class="rounded font-medium px-2 py-1 text-c2  bg-secondary dark:bg-dark-secondary text-inverse-secondary dark:text-inverse-dark-secondary "
                      >
                        TanStack Query
                      </span>
                      <span
                        class="rounded font-medium px-2 py-1 text-c2  bg-secondary dark:bg-dark-secondary text-inverse-secondary dark:text-inverse-dark-secondary "
                      >
                        Axios
                      </span>
                      <span
                        class="rounded font-medium px-2 py-1 text-c2  bg-secondary dark:bg-dark-secondary text-inverse-secondary dark:text-inverse-dark-secondary "
                      >
                        Test Automation
                      </span>
                      <span
                        class="rounded font-medium px-2 py-1 text-c2  bg-secondary dark:bg-dark-secondary text-inverse-secondary dark:text-inverse-dark-secondary "
                      >
                        Playwright
                      </span>
                      <span
                        class="rounded font-medium px-2 py-1 text-c2  bg-secondary dark:bg-dark-secondary text-inverse-secondary dark:text-inverse-dark-secondary "
                      >
                        Regex
                      </span>
                      <span
                        class="rounded font-medium px-2 py-1 text-c2  bg-secondary dark:bg-dark-secondary text-inverse-secondary dark:text-inverse-dark-secondary "
                      >
                        Data Modeling
                      </span>
                      <span
                        class="rounded font-medium px-2 py-1 text-c2  bg-secondary dark:bg-dark-secondary text-inverse-secondary dark:text-inverse-dark-secondary "
                      >
                        PostgreSQL
                      </span>
                      <span
                        class="rounded font-medium px-2 py-1 text-c2  bg-secondary dark:bg-dark-secondary text-inverse-secondary dark:text-inverse-dark-secondary "
                      >
                        Prisma
                      </span>
                      <span
                        class="rounded font-medium px-2 py-1 text-c2  bg-secondary dark:bg-dark-secondary text-inverse-secondary dark:text-inverse-dark-secondary "
                      >
                        SQL
                      </span>
                      <span
                        class="rounded font-medium px-2 py-1 text-c2  bg-secondary dark:bg-dark-secondary text-inverse-secondary dark:text-inverse-dark-secondary "
                      >
                        HTML
                      </span>
                      <span
                        class="rounded font-medium px-2 py-1 text-c2  bg-secondary dark:bg-dark-secondary text-inverse-secondary dark:text-inverse-dark-secondary "
                      >
                        Tailwind CSS
                      </span>
                      <span
                        class="rounded font-medium px-2 py-1 text-c2  bg-secondary dark:bg-dark-secondary text-inverse-secondary dark:text-inverse-dark-secondary "
                      >
                        CSS
                      </span>
                      <span
                        class="rounded font-medium px-2 py-1 text-c2  bg-secondary dark:bg-dark-secondary text-inverse-secondary dark:text-inverse-dark-secondary "
                      >
                        NextUI
                      </span>
                      <span
                        class="rounded font-medium px-2 py-1 text-c2  bg-secondary dark:bg-dark-secondary text-inverse-secondary dark:text-inverse-dark-secondary "
                      >
                        SEO
                      </span>
                      <span
                        class="rounded font-medium px-2 py-1 text-c2  bg-secondary dark:bg-dark-secondary text-inverse-secondary dark:text-inverse-dark-secondary "
                      >
                        Software Development
                      </span>
                      <span
                        class="rounded font-medium px-2 py-1 text-c2  bg-secondary dark:bg-dark-secondary text-inverse-secondary dark:text-inverse-dark-secondary "
                      >
                        Documentation
                      </span>
                      <span
                        class="rounded font-medium px-2 py-1 text-c2  bg-secondary dark:bg-dark-secondary text-inverse-secondary dark:text-inverse-dark-secondary "
                      >
                        Faker.js
                      </span>
                      <span
                        class="rounded font-medium px-2 py-1 text-c2  bg-secondary dark:bg-dark-secondary text-inverse-secondary dark:text-inverse-dark-secondary "
                      >
                        Git
                      </span>
                      <span
                        class="rounded font-medium px-2 py-1 text-c2  bg-secondary dark:bg-dark-secondary text-inverse-secondary dark:text-inverse-dark-secondary "
                      >
                        npm
                      </span>
                    </div>
                  </div>
                  <div
                    class="col-span-2 space-y-4"
                  >
                    <div
                      class="space-y-4"
                    >
                      <p
                        style="opacity: 0;"
                      >
                        In 2023, I designed the transition to the JAMstack architecture, which will gradually replace the Magento platform by APIs. This migration will result in significant gains in application performance, and will provide greater independence for content managers. For this, I created a system using Vue, Vite, Pinia, Typescript, UNOCSS+Tailwind CSS, Storybook, urql for team collaborative development. I proposed improvements to data modeling in Hygraph (GraphQL Headless CMS), to improve performance and maintain the platform's flexibility.
                         
                      </p>
                    </div>
                    <ul
                      class="list-disc list-inside pl-4"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </section>
      </div>
    </div>
  </div>
</div>
`;
